File: online-feedback-collection-system-backend/src/main/java/com/dyashin/onlinefeedbackcollectionsystem/service/AdminService.java
package com.dyashin.onlinefeedbackcollectionsystem.service;

import java.util.List;

import com.dyashin.onlinefeedbackcollectionsystem.dto.AdminDTO;
import com.dyashin.onlinefeedbackcollectionsystem.entity.Feedback;
import com.dyashin.onlinefeedbackcollectionsystem.entity.Product;

public interface AdminService {

	boolean login(AdminDTO adminDTO);

	List<Feedback> viewAllFeedback();

	List<Feedback> viewByFeedbackReviewed(boolean feedbackReviewed);

	List<Feedback> viewFeedbackByProduct(int productId);

	List<Feedback> viewFeedbackByRating(int feedbackRating);

	boolean markFeedbackAsReviewed(int feedbackId);

}

File: online-feedback-collection-system-frontend/src/service/adminService.js
import axios from "axios";

const API_BASE_URL = "http://localhost:5080/admin";

export const adminLogin = (adminData) => {
  return axios.post(`${API_BASE_URL}/login`, adminData);
};

export const viewAllFeedback = () => {
  return axios.get(`${API_BASE_URL}/viewall`);
};

export const viewFeedbackByReviewed = (feedbackReviewed) => {
  return axios.get(
    `${API_BASE_URL}/feedbackreviewed`,
    {
      params: {
        feedbackReviewed: feedbackReviewed
      }
    }
  );
};

export const viewFeedbackByProduct = (productId) => {
  return axios.get(
    `${API_BASE_URL}/viewbyproduct`,
    {
      params: {
        productId: productId
      }
    }
  );
};

export const viewFeedbackByRating = (feedbackRating) => {
  return axios.get(
    `${API_BASE_URL}/viewbyrating`,
    {
      params: {
        feedbackRating: feedbackRating
      }
    }
  );
};

export const markFeedbackAsReviewed = (feedbackId) => {
  return axios.put(
    `${API_BASE_URL}/markfeedbackasreviewed`,
    null,
    {
      params: {
        feedbackId: feedbackId
      }
    }
  );
};
